barretenberg/cpp/src/barretenberg/stdlib/primitives/biggroup/biggroup_nafs.hpp

```diff
@@ -11,6 +11,191 @@
 
 namespace bb::stdlib::element_default {
 
+template <typename C, class Fq, class Fr, class G>
+template <size_t wnaf_size>
+std::pair<uint64_t, bool> element<C, Fq, Fr, G>::compute_secp256k1_staggered_wnaf_fragment(const uint64_t fragment_u64,
+                                                                                           const uint64_t stagger,
+                                                                                           bool is_negative,
+                                                                                           bool wnaf_skew)
+{
+    // If there is not stagger then there is no need to change anyhing
+    if (stagger == 0) {
+        return std::make_pair<uint64_t, bool>((uint64_t)0, (bool)wnaf_skew);
+    }
+    int fragment = static_cast<int>(fragment_u64);
+    // Inverse the fragment if it's negative
+    if (is_negative) {
+        fragment = -fragment;
+    }
+    // If the value is positive and there is a skew in wnaf, subtract 2ˢᵗᵃᵍᵍᵉʳ. If negative and there is
+    // skew, then add
+    if (!is_negative && wnaf_skew) {
+        fragment -= (1 << stagger);
+    } else if (is_negative && wnaf_skew) {
+        fragment += (1 << stagger);
+    }
+    // If the lowest bit is zero, then set final skew to 1 and add 1 to the absolute value of the fragment
+    bool output_skew = (fragment_u64 % 2) == 0;
+    if (!is_negative && output_skew) {
+        fragment += 1;
+    } else if (is_negative && output_skew) {
+        fragment -= 1;
+    }
+
+    uint64_t output_fragment;
+    if (fragment < 0) {
+        output_fragment = static_cast<uint64_t>((int)((1ULL << (wnaf_size - 1))) + (fragment / 2 - 1));
+    } else {
+        output_fragment =
+            static_cast<uint64_t>((1ULL << (wnaf_size - 1)) - 1ULL + (uint64_t)((uint64_t)fragment / 2 + 1));
+    }
+
+    return std::make_pair<uint64_t, bool>((uint64_t)output_fragment, (bool)output_skew);
```

Comment:
I know you're actively not making code changes in this PR but this casting seems to be unnecessary..